### ISSUE TITLE
Auto logging toggle doesn't work in raids.

### ISSUE THREAD

**Issue by deNoor** [DeadlyBossMods/DeadlyBossMods#1098]
**Describe the bug**
DBM auto logging feature stopped working for actual Season 4 awakened raids (heroic, mythic). But it continues to work correctly for M+ dungeons.

**Do you have an error log of what happened?**
No LUA errors.

**To Reproduce**
1. Enter an actual raid instance.
2. There is no DBM info message about starting logging.
3. Start logging manually (/combatlog).
4. Leave the raid instance.
5. There is no message about stopping logging either.
6. Stop your logging manually (/combatlog).

**Screenshots**
My auto logging settings:
![image](https://github.com/DeadlyBossMods/DeadlyBossMods/assets/25015943/ea0c11d8-abce-4b96-94a1-48fb090e6445)

**Did you try having DeadlyBossMods as the only enabled addon and everything else (especially something like ElvUI) disabled?**
No. :)
But autologging is disabled in other addons and I've always relied on DBM only.

**Which version of DeadlyBossMods are you using?**
10.2.41

**Was it working in a previous version? If yes, which was the last good one?**
Yes. It stopped working only a ~week ago. (13-May)

**Comment by jungheeMW** [comment]
I tested it just now, and the auto-logging function is working fine for me in S4 awakened raids. Could you try again? Preferably test the feature with only DBM enabled. RaiderIO also has an autologging function, which I have disabled.

**Comment by deNoor** [comment]
Maybe happens only in **Vault of the Incarnates**.

I'm testing in Normal. After each teleportation I wait for at least 1 minute.
Tried with both other addons enabled but no other addon manages auto logging toggle and with everything except DBM modules disabled.

Entering a raid only once:
- tried to enter a fresh instance (a character appears in dragon's room).
- tried to enter a progressed instance (a character appears in intersection room).
Result: logging _is not auto started_.

Teleporting inside the raid and then exiting.
1. Entered a progressed instance → logging has not started.
2. Used a portal back to the dragon's room still inside the instance → logging has started after this teleport.
3. Exited the instance → loggin has stopped.

Leaving instance:
- a progressed raid where logging has failed to start
- a fresh instance where logging has failed to start
Result: logging starts and then ends within some seconds.
![image](https://github.com/DeadlyBossMods/DeadlyBossMods/assets/25015943/68c49093-aaf9-49e8-ae71-0003458cd707)

I have also checked that RaiderIO auto logging works correctly with Vault of the Incarnates enabling logging on enter while RaiderIO was set as the only logging toggler.

**Comment by jungheeMW** [comment]
Next week, all raids will be awakened, so could you check to see if the problem still exists? Also, do take note if the auto-logging works in the other two raids. I have been unable to reproduce this; the auto-logging works as intended for me.

Also, in your initial post you said that started manual logging by typing /combatlog. As far as I know, once you start to manual log, DBM will no longer control it until you manually stop it.

**Comment by jungheeMW** [comment]
I have just reproduced your bug in an non-Awakened normal VOTI raid. Not sure why I wasn't able to in the week that this issue was reported.

**Comment by jungheeMW** [comment]
Tested again today, RaiderIO is able to successfully enable auto combat logging upon entering the instance, but DBM cannot, so I'm able reproduce your bug.

**Comment by MysticalOS** [comment]
I could not find the issue because I only thought it was VoTI. once I realized it also affected aberrus I refined where I debugged and found the issue was the hacky new load code that had to be added to handle zone changes without loading screens (such as entering a delve in TWW or entering caverns from surface (or returning to surface from caverns).

Because DBM performed it's checks based on loading screens, this was incompatible with the new tech that changes zones without them. The problem with new code is it only runs once, not twice. loading screens ran at 1 second and 5 seconds. with the 1 second setting the IDs, and then 5 second checking IDs for auto logging.
but SPECIAL zones only run at 5. they set the Ids AND run checks at same time, except for auto logging, that runs BEFORE Ids are set casue auto logging assumed those were already set by previous check (which doesn't run for screen less loads). basically auto logging for VoTI and aberus never ran against the correct ID.

the technical reason for auto logging running before Ids changed, was because it needed to run even if ID didn't change, for mythic+ (where difficulty can change without ID change). i drycoded a new work around for that instead that hopefully resolves that while pushing the auto logger to later in the code to always run after IDs change..

I already tested and auto logging for VoTI and aberus is now fixed, but I don't know if auto logging for M+ still works anymore. that might now be broken since the new work around for that is untested.
